blockchain/src/blockchain/push.rs

```diff
@@ -40,12 +41,13 @@ impl Blockchain {
     /// Private function to push a block.
     /// Set the trusted flag to true to skip VRF and signature verifications: when the source of the
     /// block can be trusted.
-    fn do_push(
+    fn do_push<F: PostValidationHook>(
         this: RwLockUpgradableReadGuard<Self>,
         block: Block,
         trusted: bool,
         diff: Option<TrieDiff>,
         chunks: Vec<TrieChunkWithStart>,
+        post_validation_hook: &F,
```

Comment:
I opted against a closure mainly to avoid code duplication as there are only two types of hooks we currently use.
Obviously, something like `push(..., |block, result| hook.post_validation(block, result))` would have been an option, too.